File: accsaber-database/src/main/java/de/ixsen/accsaber/database/config/AccSaberSchemaFilterProvider.java
package de.ixsen.accsaber.database.config;

import org.hibernate.tool.schema.internal.DefaultSchemaFilter;
import org.hibernate.tool.schema.spi.SchemaFilter;
import org.hibernate.tool.schema.spi.SchemaFilterProvider;

public class AccSaberSchemaFilterProvider implements SchemaFilterProvider {
    @Override
    public SchemaFilter getCreateFilter() {
        return DefaultSchemaFilter.INSTANCE;
    }

    @Override
    public SchemaFilter getDropFilter() {
        return DefaultSchemaFilter.INSTANCE;
    }

    @Override
    public SchemaFilter getMigrateFilter() {
        return DefaultSchemaFilter.INSTANCE;
    }

    @Override
    public SchemaFilter getValidateFilter() {
        return AccSaberSchemaFilter.INSTANCE;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/config/AccSaberSchemaFilterProvider.kt
package de.ixsen.accsaber.database.config

import org.hibernate.tool.schema.internal.DefaultSchemaFilter
import org.hibernate.tool.schema.spi.SchemaFilter
import org.hibernate.tool.schema.spi.SchemaFilterProvider

class AccSaberSchemaFilterProvider : SchemaFilterProvider {
    override fun getCreateFilter(): SchemaFilter {
        return DefaultSchemaFilter.INSTANCE
    }

    override fun getDropFilter(): SchemaFilter {
        return DefaultSchemaFilter.INSTANCE
    }

    override fun getMigrateFilter(): SchemaFilter {
        return DefaultSchemaFilter.INSTANCE
    }

    override fun getValidateFilter(): SchemaFilter {
        return AccSaberSchemaFilter.Companion.INSTANCE
    }
}
